controllers/machinedeploymentupgrade_controller.go

```diff
@@ -57,3 +118,92 @@ func (r *MachineDeploymentUpgradeReconciler) SetupWithManager(mgr ctrl.Manager)
 		For(&anywherev1.MachineDeploymentUpgrade{}).
 		Complete(r)
 }
+
+func (r *MachineDeploymentUpgradeReconciler) reconcile(ctx context.Context, log logr.Logger, mdUpgrade *anywherev1.MachineDeploymentUpgrade) (ctrl.Result, error) {
+	// return early if machine deployment upgrade is already complete
+	if mdUpgrade.Status.Ready {
+		log.Info("All worker nodes are upgraded")
+		return ctrl.Result{}, nil
+	}
+
+	log.Info("Upgrading all worker nodes")
+	for _, machineRef := range mdUpgrade.Spec.MachinesRequireUpgrade {
+		nodeUpgrade := nodeUpgrader(machineRef, "v1.28.3-eks-1-28-9", "v3.5.9-eks-1-28-9", false)
+		if err := r.client.Get(ctx, GetNamespacedNameType(nodeUpgrade.ObjectMeta.Name, nodeUpgrade.ObjectMeta.Namespace), nodeUpgrade); err == nil {
+			if err := r.client.Update(ctx, nodeUpgrade); err != nil {
+				return ctrl.Result{}, err
+			}
+		} else if err := r.client.Create(ctx, nodeUpgrade); err != nil {
+			return ctrl.Result{}, fmt.Errorf("failed to create node upgrader for machine %s:  %v", machineRef.Name, err)
+		}
+	}
+	total := len(mdUpgrade.Spec.MachinesRequireUpgrade)
+	mdUpgrade.Status.RequireUpgrade = int64(total)
```

Comment:
I think we should keep all status related code in an `updateStatus` function so that it can be separated from the initiation of the creation of these objects.